source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.md

```diff
@@ -1263,7 +1265,7 @@ the behavior that users want. They can also result in cases where queue access o
 and not full. If a driver has a full waitQueue, then all requests for [Connections](#connection) will be rejected. If
 the client is continually spammed with requests, you could wind up with a scenario where as soon as the waitQueue is no
 longer full, it is immediately filled. It is not a favorable situation to be in, partially b/c it violates the fairness
-guarantee that the waitQueue normally provides.
+recommendation that the waitQueue normally adheres to.
```

Comment:
Does anyone understand what's written here?

> prioritizing older requests over newer requests, which is not necessarily the behavior that users want

That's literally fairness, which was required (and now will be recommended) by the spec. Yet here it is referred to as something bad.

> It is not a favorable situation to be in, partially b/c it violates the fairness
recommendation that the waitQueue normally adheres to.

I fail to understand how fairness is violated because of `waitQueueSize`. Not arguing for `waitQueueSize`, just want to understand what is written in this section. Could someone please help me?

Reply:
I believe what this is trying to say is that if you look at all requests in the "oscillates and forth between full
and not full" scenario, then we can say that those requests were not handled in a fair manner. For example, here's a specific scenario:

1. wait queue is full
1. new request comes in at time T1, it is rejected immediately because the wait queue is full
1. wait queue decreases in size
1. new request comes in at time T2, it is accepted into the wait queue.

A newer request (T2) is prioritized over an older request (T1) which violates fairness. With this in mind, I believe this is a typo and it's supposed to say "prioritizing newer requests over older requests".

Reply:
If you agree, could you add this example to help future readers?

Reply:
Thank you, @ShaneHarvey! Do you think the change made in https://github.com/mongodb/specifications/pull/1738/commits/9becee2deb1d07c9c8c42306506f3dfe72ebd136 clarifies this adequately?

Reply:
Thanks, that's great.